File: backend/src/main/java/com/stjoseph/assessmentsystem/model/Event.java
package com.stjoseph.assessmentsystem.model;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "events")
public class Event {
    @Id
    private String id;
    
    private String title;
    private String description;
    private String location;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private String organizerId; // Alumni who organized this event
    private String organizerName;
    private String organizerEmail;
    private EventStatus status;
    private EventType type;
    private String specialRequirements;
    private String rejectionReason;
    private String approvedBy; // Management ID who approved
    private String rejectedBy; // Management ID who rejected
    private LocalDateTime submittedAt;
    private LocalDateTime approvedAt;
    private LocalDateTime rejectedAt;
    private List<String> attendees; // List of user IDs who will attend
    private String department;
    private String targetAudience;
    private Integer maxAttendees;
    private String contactEmail;
    private String contactPhone;
    private String requestedBy; // Management ID who requested this event from alumni
    private String requestedByName; // Management name who requested this event from alumni
    
    public enum EventStatus {
        PENDING, APPROVED, REJECTED, CANCELLED, COMPLETED
    }
    
    public enum EventType {
        ALUMNI_INITIATED, // Event requested by alumni
        MANAGEMENT_REQUESTED // Event requested by management to alumni
    }
    
    public Event() {
        this.submittedAt = LocalDateTime.now();
        this.status = EventStatus.PENDING;
    }
    
    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }
    
    public LocalDateTime getStartDateTime() { return startDateTime; }
    public void setStartDateTime(LocalDateTime startDateTime) { this.startDateTime = startDateTime; }
    
    public LocalDateTime getEndDateTime() { return endDateTime; }
    public void setEndDateTime(LocalDateTime endDateTime) { this.endDateTime = endDateTime; }
    
    public String getOrganizerId() { return organizerId; }
    public void setOrganizerId(String organizerId) { this.organizerId = organizerId; }
    
    public String getOrganizerName() { return organizerName; }
    public void setOrganizerName(String organizerName) { this.organizerName = organizerName; }
    
    public String getOrganizerEmail() { return organizerEmail; }
    public void setOrganizerEmail(String organizerEmail) { this.organizerEmail = organizerEmail; }
    
    public EventStatus getStatus() { return status; }
    public void setStatus(EventStatus status) { this.status = status; }
    
    public EventType getType() { return type; }
    public void setType(EventType type) { this.type = type; }
    
    public String getSpecialRequirements() { return specialRequirements; }
    public void setSpecialRequirements(String specialRequirements) { this.specialRequirements = specialRequirements; }
    
    public String getRejectionReason() { return rejectionReason; }
    public void setRejectionReason(String rejectionReason) { this.rejectionReason = rejectionReason; }
    
    public String getApprovedBy() { return approvedBy; }
    public void setApprovedBy(String approvedBy) { this.approvedBy = approvedBy; }
    
    public String getRejectedBy() { return rejectedBy; }
    public void setRejectedBy(String rejectedBy) { this.rejectedBy = rejectedBy; }
    
    public LocalDateTime getSubmittedAt() { return submittedAt; }
    public void setSubmittedAt(LocalDateTime submittedAt) { this.submittedAt = submittedAt; }
    
    public LocalDateTime getApprovedAt() { return approvedAt; }
    public void setApprovedAt(LocalDateTime approvedAt) { this.approvedAt = approvedAt; }
    
    public LocalDateTime getRejectedAt() { return rejectedAt; }
    public void setRejectedAt(LocalDateTime rejectedAt) { this.rejectedAt = rejectedAt; }
    
    public List<String> getAttendees() { return attendees; }
    public void setAttendees(List<String> attendees) { this.attendees = attendees; }
    
    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }
    
    public String getTargetAudience() { return targetAudience; }
    public void setTargetAudience(String targetAudience) { this.targetAudience = targetAudience; }
    
    public Integer getMaxAttendees() { return maxAttendees; }
    public void setMaxAttendees(Integer maxAttendees) { this.maxAttendees = maxAttendees; }
    
    public String getContactEmail() { return contactEmail; }
    public void setContactEmail(String contactEmail) { this.contactEmail = contactEmail; }
    
    public String getContactPhone() { return contactPhone; }
    public void setContactPhone(String contactPhone) { this.contactPhone = contactPhone; }
    
    public String getRequestedBy() { return requestedBy; }
    public void setRequestedBy(String requestedBy) { this.requestedBy = requestedBy; }
    
    public String getRequestedByName() { return requestedByName; }
    public void setRequestedByName(String requestedByName) { this.requestedByName = requestedByName; }
}

File: lib/models/event.dart
class Event {
  final String id;
  final String title;
  final String description;
  final String location;
  final DateTime startDateTime;
  final DateTime? endDateTime;
  final String organizerId;
  final String organizerName;
  final String? organizerEmail;
  final EventStatus status;
  final EventType type;
  final String? specialRequirements;
  final String? rejectionReason;
  final String? approvedBy;
  final String? rejectedBy;
  final DateTime submittedAt;
  final DateTime? approvedAt;
  final DateTime? rejectedAt;
  final List<String> attendees;
  final String? department;
  final String? targetAudience;
  final int? maxAttendees;
  final String? contactEmail;
  final String? contactPhone;
  final String? requestedBy;
  final String? requestedByName;

  const Event({
    required this.id,
    required this.title,
    required this.description,
    required this.location,
    required this.startDateTime,
    this.endDateTime,
    required this.organizerId,
    required this.organizerName,
    this.organizerEmail,
    required this.status,
    required this.type,
    this.specialRequirements,
    this.rejectionReason,
    this.approvedBy,
    this.rejectedBy,
    required this.submittedAt,
    this.approvedAt,
    this.rejectedAt,
    required this.attendees,
    this.department,
    this.targetAudience,
    this.maxAttendees,
    this.contactEmail,
    this.contactPhone,
    this.requestedBy,
    this.requestedByName,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      location: json['location'] ?? '',
      startDateTime: DateTime.parse(json['startDateTime'] ?? DateTime.now().toIso8601String()),
      endDateTime: json['endDateTime'] != null ? DateTime.parse(json['endDateTime']) : null,
      organizerId: json['organizerId'] ?? '',
      organizerName: json['organizerName'] ?? '',
      organizerEmail: json['organizerEmail'],
      status: EventStatus.values.firstWhere(
        (e) => e.name.toUpperCase() == (json['status'] ?? 'PENDING').toString().toUpperCase(),
        orElse: () => EventStatus.PENDING,
      ),
      type: EventType.values.firstWhere(
        (e) => e.name.toUpperCase() == (json['type'] ?? 'ALUMNI_INITIATED').toString().toUpperCase(),
        orElse: () => EventType.ALUMNI_INITIATED,
      ),
      specialRequirements: json['specialRequirements'],
      rejectionReason: json['rejectionReason'],
      approvedBy: json['approvedBy'],
      rejectedBy: json['rejectedBy'],
      submittedAt: DateTime.parse(json['submittedAt'] ?? DateTime.now().toIso8601String()),
      approvedAt: json['approvedAt'] != null ? DateTime.parse(json['approvedAt']) : null,
      rejectedAt: json['rejectedAt'] != null ? DateTime.parse(json['rejectedAt']) : null,
      attendees: json['attendees'] != null ? List<String>.from(json['attendees']) : [],
      department: json['department'],
      targetAudience: json['targetAudience'],
      maxAttendees: json['maxAttendees'],
      contactEmail: json['contactEmail'],
      contactPhone: json['contactPhone'],
      requestedBy: json['requestedBy'],
      requestedByName: json['requestedByName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'location': location,
      'startDateTime': startDateTime.toIso8601String(),
      'endDateTime': endDateTime?.toIso8601String(),
      'organizerId': organizerId,
      'organizerName': organizerName,
      'organizerEmail': organizerEmail,
      'status': status.name.toUpperCase(),
      'type': type.name.toUpperCase(),
      'specialRequirements': specialRequirements,
      'rejectionReason': rejectionReason,
      'approvedBy': approvedBy,
      'rejectedBy': rejectedBy,
      'submittedAt': submittedAt.toIso8601String(),
      'approvedAt': approvedAt?.toIso8601String(),
      'rejectedAt': rejectedAt?.toIso8601String(),
      'attendees': attendees,
      'department': department,
      'targetAudience': targetAudience,
      'maxAttendees': maxAttendees,
      'contactEmail': contactEmail,
      'contactPhone': contactPhone,
      'requestedBy': requestedBy,
      'requestedByName': requestedByName,
    };
  }
}

enum EventStatus {
  PENDING,
  APPROVED,
  REJECTED,
  CANCELLED,
  COMPLETED,
}

enum EventType {
  ALUMNI_INITIATED,
  MANAGEMENT_REQUESTED,
}

extension EventStatusExtension on EventStatus {
  String get displayName {
    switch (this) {
      case EventStatus.PENDING:
        return 'Pending';
      case EventStatus.APPROVED:
        return 'Approved';
      case EventStatus.REJECTED:
        return 'Rejected';
      case EventStatus.CANCELLED:
        return 'Cancelled';
      case EventStatus.COMPLETED:
        return 'Completed';
    }
  }

  Color get color {
    switch (this) {
      case EventStatus.PENDING:
        return const Color(0xFFF59E0B);
      case EventStatus.APPROVED:
        return const Color(0xFF059669);
      case EventStatus.REJECTED:
        return const Color(0xFFDC2626);
      case EventStatus.CANCELLED:
        return const Color(0xFF6B7280);
      case EventStatus.COMPLETED:
        return const Color(0xFF2563EB);
    }
  }
}